tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/BulkheadTest.java

```diff
@@ -0,0 +1,309 @@
+    private ExecutorService xService = Executors.newFixedThreadPool(THREADPOOL_SIZE);
+
+    @Inject
+    private BulkheadClassSemaphoreDefaultBean bhBeanClassSemaphoreDefault;
+    @Inject
+    private BulkheadMethodSemaphoreDefaultBean bhBeanMethodSemaphoreDefault;
+    @Inject
+    private BulkheadClassAsynchronousDefaultBean bhBeanClassAsynchronousDefault;
+    @Inject
+    private BulkheadMethodAsynchronousDefaultBean bhBeanMethodAsynchronousDefault;
+
+    @Inject
+    private BulkheadClassSemaphore10Bean bhBeanClassSemaphore10;
+    @Inject
+    private BulkheadMethodSemaphore10Bean bhBeanMethodSemaphore10;
+    @Inject
+    private BulkheadClassAsynchronous10Bean bhBeanClassAsynchronous10;
+    @Inject
+    private BulkheadMethodAsynchronous10Bean bhBeanMethodAsynchronous10;
+
+    @Inject
+    private BulkheadClassAsynchronousQueueingBean bhBeanClassAsynchronousQueueing;
+    @Inject
+    private BulkheadMethodAsynchronousQueueingBean bhBeanMethodAsynchronousQueueing;
+
+    
+    
+    
+    @Deployment
+    public static WebArchive deploy() {
+        JavaArchive testJar = ShrinkWrap.create(JavaArchive.class, "ftBulkheadTest.jar")
+                .addPackage(BulkheadClassSemaphore10Bean.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml").as(JavaArchive.class);
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "ftBulkheadTest.war").addAsLibrary(testJar);
+        return war;
+    }
+
+    @BeforeTest
+    public void reset() {
+        Checker.reset();
+    }
+
+    public void check() {
+        Checker.check();
+    }
+
+    /**
+     * Tests the class synchronous Bulkhead10.
+     */
+    @Test()
+    public void testBulkheadClassSemaphore10() {
+        threads(20, bhBeanClassSemaphore10, 10);
+        check();
+
+    }
+
+    /**
+     * Tests the method synchronous Bulkhead10.
+     */
+    @Test()
+    public void testBulkheadMethodSemaphore10() {
+        threads(20, bhBeanMethodSemaphore10, 10);
+        check();
+    }
+
+    /**
+     * Tests the class asynchronous Bulkhead10.
+     */
+    @Test()
+    public void testBulkheadClassAsynchronous10() {
+        loop(10, bhBeanClassAsynchronous10, 10);
+        check();
+    }
+
+    /**
+     * Tests the method asynchronous Bulkhead(10).
+     */
+    @Test()
+    public void testBulkheadMethodAsynchronous10() {
+        loop(10, bhBeanMethodAsynchronous10, 10);
+        check();
+    }
+
+    /**
+     * Tests the basic class synchronous Bulkhead.
+     */
+    @Test()
+    public void testBulkheadClassSemaphoreDefault() {
+        threads(20, bhBeanClassSemaphoreDefault, 10);
+        check();
+
+    }
+
+    /**
+     * Tests the basic method synchronous Bulkhead.
+     */
+    @Test()
+    public void testBulkheadMethodSemaphoreDefault() {
+        threads(20, bhBeanMethodSemaphoreDefault, 10);
+        check();
+    }
+
+    /**
+     * Tests the basic class synchronous Bulkhead.
```

Comment:
Do you mean async? Can you add some more comments to explain each test instead of just saying basic method synchronous test?